src/main/java/org/testcontainers/containercore/Container.java

```diff
@@ -0,0 +1,39 @@
+package org.testcontainers.containercore;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * TODO: Javadocs
+ */
+public class Container {
```

Comment:
I would like to suggest making this class `AutoClosable`. The interface can simply invoke `stop` and should override the method to remove the `Exception` as none is thrown.

This allows for a very comfortable usage of containers such as:

try(Container container = new MyContainer()) {
  container.start();
  // do something with container..
}

Reply:
@raphw 👍  good catch! In fact, we already have it in TestContainers and it proved to be useful :)

Reply:
👍 absolutely - a small detail that adds a lot of benefit. Thanks!